sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Security/Ucfg/UcfgBuilder_Instructions.cs

```diff
@@ -787,6 +787,58 @@ public void Foo(Class1 other)
 
         }
     }
+}";
+            UcfgVerifier.VerifyInstructions(code, "Foo");
+        }
+        
+        [TestMethod]
+        public void ArrayCreation_WithNew()
+        {
+            const string code = @"
+namespace Namespace
+{
+    public class Class1
+    {
+        public void Foo()
+        {
+            // Primitive type
+            var x0 = new string[42];    // %0 := new string[]
+                                        // x0 := __id [ %0 ]
+
+            // Array of objects
+            var x1 = new Class1[0];     // %1 := new Namespace.Class1[]
+                                        // x1 := __id [ %1 ]
+
+            // Multi-rank arrays
+            var x2 = new int[10,2];     // %2 := new int[*,*]
+                                        // x2 := __id [ %2 ]
+
+            // Arrau with initializer (initializer is ignored)
+            var x3 = new string[] { ""aaa"", ""bbb"", ""ccc"" };    // %3 := new string[]
+                                                                    // x3 := __id [ %3 ]
+        }
+    }
+}";
+
+            var x3 = new string[] { "aaa", "bbb", "ccc" };
+
+            UcfgVerifier.VerifyInstructions(code, "Foo");
+        }
+
+        [TestMethod] [Ignore] // TODO: this method currently throws a null-ref
+        public void ArrayCreation_WithCreateInstance()
+        {
+            const string code = @"
+namespace Namespace
+{
+    public class Class1
+    {
+        public void Foo()
+        {
+            var a = System.Array.CreateInstance(typeof(string), 10);    // %0 = System.Array.CreateInstance(System.Type, int) [ System.Array.CreateInstance ]

```

Comment:
I thought the problem was the _typeof_ but it isn't getting that far.
Currently the code is failing when it tries to treat the namespace syntax node for _System_ as a named-type node.